Kudu.Core/Jobs/ContinuousJobRunner.cs

```diff
@@ -297,5 +302,26 @@ protected virtual void Dispose(bool disposing)
                 }
             }
         }
+
+        private int GetAvailablePortForWebJob()
+        {
+            if (JobSettings.IsWebHook)
+            {
+                var listener = new TcpListener(IPAddress.Loopback, 0);
+                try
+                {
+                    listener.Start();
+                    return ((IPEndPoint)listener.LocalEndpoint).Port;
+                }
+                finally
+                {
+                    listener.Stop();
```

Comment:
It's pretty sequential how the OS picks a port number, but I think it's sequential OS wide so I don't think 2 Kudus running on the same machine could get the same port number. 
Also a job in a different site won't be able to steal the request because the sandbox doesn't allow communicating with local ports outside the sandbox so there is no security issue. The only real 100% guaranteed alternative is for kudu to wait for the webjob to pick a random port and then communicate that port back to kudu.